src/read_params.jl

```diff
@@ -4,47 +4,130 @@
 
 using ACE1pack
 
-export  fill_defaults!
+export  fill_defaults!, parse_basis_keys
 
-""" Recursively updates nested dictionaries with default parameters"""
+""" 
+Recursively updates nested dictionaries with default parameters. 
+One only needs to suply the mandatory parameters, for example
+````
+minimal_params = Dict(
+    "data" => Dict(
+        "fname" => "something"),
+    "basis" => Dict(
+        "rpi_basis" => Dict(
+            "species" => "something",
+            "N" => 1,
+            "maxdeg" => 1
+            ),
+        "pair_basis" => Dict(
+            "species" => "sth",
+            "maxdeg" => 1
+            ),),
+    "solver" => Dict(
+        "solver" => "rrqr"),
+    "e0" => "something")
+```
+"""
 function fill_defaults!(params::Dict; param_key = "fit_params")
-    #TODO: "transforms" actually need processing, but it's less straightforward
-    dicts_to_not_process = ["weights", "e0", "transforms", "cutoffs", "Dd", "Dn", "Dl"]
     # Go through the nested dictionaries filling in the default values
     params = _fill_default(params, param_key)
     for (key, val) in params
         if key == "basis"
             for (basis_name, basis_params) in params[key]
-                params[key][basis_name] = fill_defaults!(basis_params; param_key=key)
+                if basis_name == "rpi"
+                    basis_params = parse_basis_keys(basis_param)
+                end
+                params[key][basis_name] = fill_defaults!(basis_params; param_key=basis_name)
             end
-        elseif val isa Dict &&  ~(key in dicts_to_not_process)
+        elseif key == "transforms"
+            params[key] = _fill_transforms(val)
+        elseif val isa Dict && haskey(_dict_constructors, key)
             params[key] = fill_defaults!(val; param_key = key)
         end
     end
     return params
 end
 
-"""
-Converts dictionary of parameters to keyword arguments and
-calls ACE1pack parameter constructor functions
-"""
-function _fill_default(d::Dict, key)
-    dict_constructor = _dict_constructors[key]
-    kwargs = _params_to_kwargs(d)
-    return dict_constructor(;kwargs...)
-end
+_fill_transforms(params) = Dict([key => fill_defaults!(val, param_key="transform") for (key, val) in params])
+_fill_default(d::Dict, key) = _dict_constructors[key](;_makesymbol(d)...)
+_makesymbol(p::Pair) = (Symbol(p.first) => (p.second))
+_makesymbol(D::Dict) = Dict(_makesymbol.([D...])...)   
 
 _dict_constructors = Dict(
     "fit_params" => ACE1pack.fit_params,
     "data" => ACE1pack.data_params,
     "solver" => ACE1pack.solver_params,
-    "basis" => ACE1pack.basis_params,
-    "rad_basis" => ACE1pack.basis_params,
-    "transform" => ACE1pack.transform_params,
+    "rad_basis" => ACE1pack.rad_basis_params,
+    "rpi_basis" => ACE1pack.rpi_basis_params,
+    "pair_basis" => ACE1pack.pair_basis_params,
+    "transform" => ACE1pack.transform_params, 
```

Comment:
Are the basis and transform params docstrings coming?

Reply:
They should be there - how do `?basis_params` and `?transform_params` look?

None of named basis params (e.g. `rad_basis_params`) are exported, because the default parameters' constructor should be `basis_params` with the `type` specified.

Reply:
How do I view them so things are linked? I've just been looking that the github diffs, which just show `?basis_params`.

Reply:
Can you explain? The docstrings are with `basis_params()` and `transform_params()` in `basis.jl`. They are printed when `?basis_params` or `transform_params` get called from the Julia REPL.

Reply:
Not sure; why wouldn't the non-processed docstrings in basis.jl work for you?

btw - maybe you'll find this useful until [JuliaMolSim/MolSim#4](https://github.com/JuliaMolSim/MolSim/pull/4) gets resolved. I think I got around that issue with
* delete ACE1pack
* Install and dev IPFitting 
* pull from the correct branch - https://github.com/ACEsuit/IPFitting.jl, not the outdated  "https://github.com/cortner/IPFitting.jl.git" which is accessed by default with "dev IPFitting"
* Re-install ACE1pack. 

Might still be a problem with ACE1pack asking for v0.10.1 of IPFitting and the current main of IPFitting being on v0.10.0, but I don't remember that causing (too much) trouble...

Reply:
Just github's silliness of not showing regions when too much has changed.  I did a search for the relevant strings, and found nothing, because that section of diffs was collapsed.  I'll look now.